pkg/controllers/cloud/account_poller.go

```diff
@@ -217,15 +227,50 @@ func (p *Poller) updateAccountPoller(name *types.NamespacedName, selector *cloud
 	return nil
 }
 
-func (p *accountPoller) doAccountPoller() {
+// RestartAccountPoller stops and starts a goroutine making sure there is only one account poller goroutine at a time.
+func (p *Poller) RestartAccountPoller(name *types.NamespacedName) error {
```

Comment:
nit: As it will be accessed within package, change to restartAccountPoller

Reply:
done